core/src/main/java/edu/wpi/grip/core/metrics/Analysis.java

```diff
@@ -0,0 +1,102 @@
+package edu.wpi.grip.core.metrics;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Analysis data of a timed action. Contains
+ * <ul>
+ * <li>The number of samples taken</li>
+ * <li>The average of the samples taken. This may not be an arithmetic mean, or even an average
+ * of <i>all</i> the samples</li>
+ * <li>Statistical analysis of the data.</li>
+ * </ul>
+ */
+@Immutable
+public final class Analysis {
+
+  private static final int windowSize = 10;
```

Comment:
All caps with underscores so that it's clearly a static final variable.
(This should probably be in checkstyle).

Also, can you document what this means?